src/pps/pps.proto

```diff
@@ -117,6 +118,7 @@ message PFSInput {
 message CronInput {
   string name = 1;
   string repo = 2;
+  string project = 7;
```

Comment:
I see what you mean — what is the use case for a pipeline in one project to create a repo in another project — but isn’t that an artificial limitation?

Reply:
I don't totally understand the question. Restricting this can be called "artificial", but I don't think we need to unnecessarily expand our behavior surface.

Reply:
What I mean is:

- Every pipeline is in a project.
- Every pipeline creates an output repo in its project.
- Every pipeline has inputs.
  - PFS input repos are able to exist outside the pipeline’s project.
  - By analogy I expect cron input repos to be able to exist outside the pipeline’s project.

It just seems like a really weird special case: ‘PFS input repos can belong to other projects, but not cron input repos.’ Cron inputs _are_ a bit different from PFS inputs: PFS inputs already exist, while cron inputs are created implicitly, and it _is_ a bit strange for a user to create the input repo in a different project, but it seems even stranger (to me!) to prevent him.

Particularly because he could always just create a pipeline with a cron input in one project, and use its output repo as the input to an identity pipeline in another project. Disallowing this in the first place doesn’t prevent anything; it just makes it more awkward, and forces users to run two pipelines instead of one.

My intuition here is that in the codebase before projects, repos were always identified by a simple key, {name}; now they are identified by a compound key {project, name}, thus any message which refers to a repo by name now needs to refer to it by project and name. It seems inconsistent for every message _but_ the cron input message to add the project, while cron input messages infer their projects from some enclosing message. It’s even more inconsistent, because sometimes a cron input message is contained in a job info details message; sometimes in a pipeline info details message; sometimes in a list datum request; and sometimes in a create pipeline request. So anything which needs to handle cron input messages and their repos would need to be changed to take another parameter for the project.

Reply:
I think most of the source of struggle is in the fact that unlike typical pipelines that imply an output repo, cron pipelines also imply an input repo. Because they're intentionally tightly coupled to the pipeline, I don't see a good argument to say we need to support the case where the cron repo is in a separate project. Like you said, if that becomes a use-case the user can always create a cron pipeline in a separate project that feeds into a pipeline in another. 

Another component to the argument is that we shouldn't do more work to support unlikely use-cases.

Reply:
@robert-uhl asked for my input here. I would lean towards including the project, particularly if it makes the implementation simpler. I imagine having that information readily available will be easier to work with than inferring it. But, longer term, I would lean more towards the model @acohen4 is getting at, which is that cron inputs should be more tightly coupled to pipelines. I think we would need to take the bigger step of making them system repos and rethinking the way we specify cron inputs for that. So, for now, I think we should go with what is consistent rather than take a half step towards the better long term model.

Reply:
To respect the implied API contract, we would now need extra logic on pipeline creation to figure out whether the project exists and whether the user has permissions to create a pipeline in the project. Not a big deal, but I think I would need a user argument to agree that this is worth doing. My concern is that this would be adding tech debt. 

I'm also not sure I'm aligned with the consistency argument. There are already several non-overlapping fields between PFSInput and CronInput (and the other input types). That seems like a product of them having different functionality.

I also realize this is just adding one field, so it probably isn't the most consequential decision.